components/src/components/Modal.tsx

```diff
@@ -0,0 +1,87 @@
+/** @jsxImportSource @emotion/react */
+
+import styled from "@emotion/styled";
+import { CloseIcon } from "./common";
+import { PropsWithChildren } from "react";
+import { css } from "@emotion/react";
+
+const ModalContainer = styled.div<{ position: "center" | "bottom" }>`
+  width: 280px;
+  height: fit-content;
+
+  background-color: white;
+  padding: 24px 32px;
+
+  border-radius: ${(props) =>
+    props.position === "center" ? "8px" : "8px 8px 0 0"};
+
+  position: fixed;
+
+  left: 0;
+  right: 0;
+  margin: 0 auto;
+
+  top: ${(props) => (props.position === "center" ? "50%" : "auto")};
+  transform: translateY(-50%);
+
+  bottom: 0;
+
+  z-index: 2;
+`;
+
+const ModalTop = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+`;
+
+const Title = styled.div`
+  font-size: 18px;
+  font-weight: 700;
+`;
+
+const ModalBackdrop = styled.div`
+  background-color: #000;
+  opacity: 35%;
+
+  position: fixed;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+  margin: auto;
+`;
+
+interface ModalInterface {
+  position?: "center" | "bottom";
+  title: string;
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+export default function Modal({
+  position = "center",
+  title,
+  onClose,
+  children,
+  isOpen,
+}: PropsWithChildren<ModalInterface>) {
+  if (!isOpen) return;
+  return (
+    <>
+      <ModalContainer position={position}>
+        <ModalTop>
+          <Title>{title}</Title>
+          <CloseIcon onClick={onClose} css={closeIconStyle} />
+        </ModalTop>
```

Comment:
`renderHeader` 처럼 많이 유연한 props를 뚫으면, 기존의 스타일을 제안하면서도 유연한 커스텀이 가능해집니다!

```suggestion
        {
          renderHeader?.() ?? (
            <ModalTop>
              <Title>{title}</Title>
              <CloseIcon onClick={onClose} css={closeIconStyle} />
            </ModalTop>
          )
        }
```

Reply:
ddef1326b99d9d7c8ff40b91f941523193c12482

너무 조은데요
버튼 등도 이런식으로 추가할 수 있을 것 같은데 혹시 step2에서 해두 대나요,,?🥺

Reply:
넵!! 여러 미션에 적용해보면서 직접 대응하는 것이 좋아보이네요~!